README.md

```diff
@@ -1,55 +1,42 @@
-# aws-application-deployment-template
+# synapse-docker-registry
 
-CDK-based template for deploying a containerized application to AWS
+CDK-based template for deploying the Synapse Docker registry
 
-## Perequisites
 
-AWS CDK projects require some bootstrapping before synthesis or deployment.
-Please review the [bootstapping documentation](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_bootstrap)
-before development.
+## Configuration
 
-## Development
-
-The `cdk.json` file tells the CDK Toolkit how to execute your app.
-
-This project is set up like a standard Python project.  The initialization
-process also creates a virtualenv within this project, stored under the `.venv`
-directory.  To create the virtualenv it assumes that there is a `python3`
-(or `python` for Windows) executable in your path with access to the `venv`
-package. If for any reason the automatic creation of the virtualenv fails,
-you can create the virtualenv manually.
-
-To manually create a virtualenv on MacOS and Linux:
-
-```
-$ python3 -m venv .venv
-```
+We use the [AWS Secrets Manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/intro.html)
 
-After the init process completes and the virtualenv is created, you can use the following
-step to activate your virtualenv.
+In dev the secret is named `registry-dev-DockerFargateStack/dev/ecs` and in the prod stack,
+`registry-prod-DockerFargateStack/prod/ecs`
 
-```
-$ source .venv/bin/activate
-```
+A secret is a collection of key-value pairs.  For this application there is just one pair.  The key should be `notification_auth` and the value is the
+Base64 encoded "Basic auth" credentials which are a shared-secret with Synapse as the event notification recipient.
 
-If you are a Windows platform, you would activate the virtualenv like this:
+### Registry container
+We use the open source Docker `registry`, available on DockerHub.  This container requires several configuration files to be mounted.
+To achieve this cleanly, we embed the files into our own copy of the registry, published to GitHub as `ghcr.io/synapse-docker-registry`, making two versions under
+two different tags, `dev` and `prod`, each with its own configuration.  Note that the configuration files we add contain no secrets.  The only required secret (the event listener
+authorization credential) is inserted into the registry configuration file at the time the container is started up.
```

Comment:
i don't see how the secret is `inserted into the registry configuration file`.  could you please explain?

Reply:
Yes, it's line 4 in /startup.sh.  (I added this to the README to clarify.) So the secret (pulled from Secrets Manager) is passed to the ECS as an env var at run time and the container inserts it into config.yml before starting up the registry server.

Reply:
i see, it wasn't clear to me how `$notification_auth` was passed in or where it came from.  some docs or extra comments around this would definitely help.

Reply:
I just added some more explanation to `startup.sh`.